File: src/main/java/br/com/barbearia/api/service/ServicoService.java
package br.com.barbearia.api.service;

import br.com.barbearia.api.dto.ServicoRequestDTO;
import br.com.barbearia.api.dto.ServicoResponseDTO;

import java.util.List;

public interface ServicoService {

    ServicoResponseDTO criar(ServicoRequestDTO dto);

    List<ServicoResponseDTO> listar();

    ServicoResponseDTO buscarPorId(Long id);

    ServicoResponseDTO atualizar(Long id, ServicoRequestDTO dto);

    void deletar(Long id);
}

File: services/servicoService.js
import prisma from "../prisma.js";

const servicoService = {

  async criar(data) {
    return await prisma.servico.create({
      data
    });
  },

  async listar() {
    return await prisma.servico.findMany();
  },

  async buscarPorId(id) {
    return await prisma.servico.findUnique({
      where: { id: Number(id) }
    });
  },

  async atualizar(id, data) {
    return await prisma.servico.update({
      where: { id: Number(id) },
      data
    });
  },

  async deletar(id) {
    return await prisma.servico.delete({
      where: { id: Number(id) }
    });
  }

};

export default servicoService;